packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -1108,43 +1137,58 @@ export function simplify<T>(
 					const key = typeof field.key === 'string'
 						? field.key
 						: simplify(field.key, options, parents)
-					if (typeof key !== 'string' && key.kind === 'union') {
-						for (const subKey of key.members) {
-							fields.push({
-								...field,
-								key: simplifyKey(subKey),
-							})
-						}
-					} else {
-						fields.push({
-							...field,
-							key: simplifyKey(key),
-						})
-					}
+					addField(key, field)
 				} else {
 					const simplifiedSpreadType = simplify(
 						field.type,
 						options,
 						parents,
 					)
-
 					if (simplifiedSpreadType.kind === 'struct') {
-						fields.push(...simplifiedSpreadType.fields)
+						for (const field of simplifiedSpreadType.fields) {
+							addField(field.key, field)
+						}
+					} else {
+						const type = McdocType.toString(simplifiedSpreadType)
+						options.context.logger.warn(
+							`Tried to spread non-struct type ${type}`,
+						)
 					}
 				}
 			}
+			complexFields = complexFields.filter(other =>
```

Comment:
Not sure we really need this one. I wouldn't expect a complex type to be assignable to a literal.

Reply:
Fair. I will take it out but leave a comment in case we come across situations where we need it.